src/phasorpy/_phasorpy.pyx

```diff
@@ -1512,3 +1512,105 @@ cdef float_t _blend_lighten(
     if isnan(b) or isnan(a):
         return a
     return <float_t> max(a, b)
+
+###############################################################################
+# Threshold ufuncs
+
+
+@cython.ufunc
+cdef (double, double, double) _phasor_threshold(
+    float_t mean,
+    float_t real,
+    float_t imag,
+    float_t mean_min,
+    float_t mean_max,
+    float_t real_min,
+    float_t real_max,
+    float_t imag_min,
+    float_t imag_max,
+    float_t phase_min,
+    float_t phase_max,
+    float_t modulation_min,
+    float_t modulation_max,
+    float_t fill,
+) noexcept nogil:
+    """Return thresholded values."""
+    cdef:
+        double phi = NAN
+        double mod = NAN
+
+    if isnan(mean) or isnan(real) or isnan(imag):
+        return <float_t> NAN, <float_t> NAN, <float_t> NAN
```

Comment:
Should this return `fill` instead? On the other hand, I'm also not sure when one would want to set the intensity mean and phasor coordinates to the same value other than NaN or maybe zero. Zero could be risky as we have discussed, and the user could easily use [numpy.nan_to_num](https://numpy.org/doc/stable/reference/generated/numpy.nan_to_num.html) to replace NaNs with anything.

Reply:
Good question. I'm not sure, but I think is better to propagate the `NaN` rather than the fill value. Although if that is the case it should be documented somewhere in the docstring. You are also right that replacing the `NaN`s  should be easy. Maybe at this stage is better to remove the `fill_value` as you said and see if it is necessary in the future. I will update and fill with `NaN`.